apollo-runtime/src/main/java/com/apollographql/apollo/internal/ApolloCallTracker.java

```diff
@@ -0,0 +1,175 @@
+package com.apollographql.apollo.internal;
+
+import com.apollographql.apollo.ApolloCall;
+import com.apollographql.apollo.ApolloPrefetch;
+import com.apollographql.apollo.IdleResourceCallback;
+
+import java.util.LinkedHashSet;
+import java.util.Set;
+
+/**
+ * ApolloCallTracker is responsible for keeping track of running {@link ApolloCall} & {@link ApolloPrefetch} objects.
+ */
+public final class ApolloCallTracker {
+
+  private IdleResourceCallback idleResourceCallback;
```

Comment:
I would not use Runnable for this as this is really callback not some runnable action that we need to run.